File: solutions/alien-language/dart/approach3.dart
import 'dart:io';

main() {

  new File('A-large-practice.in').readAsLines().then((List<String> lines) {
    var ldn = lines[0].split(" ");
    int D = int.parse(ldn[1]); 
    int N = int.parse(ldn[2]);
    
    var palabras = [];
    var patterns = [];

    for(int i=0; i<D; i++){
      palabras.add(lines[i+1]);
    }

    for(int i=0; i < N; i++){
      patterns.add(lines[i+D+1]);
      patterns[i] = patterns[i].replaceAll("(","[").replaceAll(")","]");
    }

    for(int i = 0; i < N; i++){
      int cont = 0;
      var pattern = new RegExp(patterns[i]);
      for(int j = 0; j < D; j++){
        if(pattern.hasMatch(palabras[j])){
          cont++;
        }
      }
      print("Case #${i+1}: ${cont}");
    }
  });
}

File: solutions/alien-language/go/approach3/approach3.go
package main

import (
    "fmt"
    "regexp"
    "strings"
    "strconv"
    "log"
 	"bufio"
 	"os"
)

func main(){

	file, err := os.Open("A-large-practice.in")
    if err != nil {
        log.Fatal(err)
    }   
    defer file.Close()
    scanner := bufio.NewScanner(file)
    scanner.Scan();
    
    ldn := strings.Split(scanner.Text(), " ")
    fmt.Println(ldn)

    D, _ := strconv.Atoi(ldn[1])
    N, _ := strconv.Atoi(ldn[2])

    var palabras []string
	var patterns []string

    palabras = make([]string, D) 
	patterns = make([]string, N)

    for i := 0; i < D; i++ {	
		scanner.Scan()
		palabras[i] = scanner.Text()
	}

	for i := 0; i < N; i++ {	
		scanner.Scan()
		patterns[i] = scanner.Text()
	}

    for i := 0; i < len(patterns); i++ {
		
		cont := 0
    	pattern := strings.Replace((strings.Replace(patterns[i], "(", "[", -1)), ")", "]", -1)
	    
	    for j := 0; j < len(palabras); j++ {	
		
			match, _ := regexp.MatchString(pattern, palabras[j])
    	
    		if match {
    	
    			cont++
   		
   		 	}
		
		}

		fmt.Println("Case #"+strconv.Itoa(i+1)+": "+strconv.Itoa(cont))
	}
}


 

    

